File: backend-spring/src/main/java/org/lafeuille/demo/data/EventRepository.java
package org.lafeuille.demo.data;

import java.util.UUID;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EventRepository extends JpaRepository<Event, UUID> {
  Page<Event> findAllByCalendarId(UUID calendarId, Pageable pageable);
}

File: backend-spring/src/main/kotlin/org/lafeuille/demo/data/EventRepository.kt
package org.lafeuille.demo.data

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface EventRepository : JpaRepository<Event, UUID> {
    fun findAllByCalendarId(
        calendarId: UUID,
        pageable: Pageable,
    ): Page<Event>
}
